appshell/cef_main_window.cpp

```diff
@@ -99,6 +99,60 @@ LPCWSTR cef_main_window::GetBracketsWindowTitleText()
     return szTitle;
 }
 
+void cef_main_window::EnsureWindowRectVisibility(int& left, int& top, int& width, int& height, int showCmd)
+{
+    static const int kWindowFrameSize = 8;
+
+    // don't check if we're already letting
+    //  Windows determine the window placement
+    if (left   == CW_USEDEFAULT &&
+        top    == CW_USEDEFAULT &&
+        width  == CW_USEDEFAULT &&
+        height == CW_USEDEFAULT) {
+            return;
+    }
+
+    // The virtual display is the bounding rect of all monitors
+    // see http://msdn.microsoft.com/en-us/library/dd162729(v=vs.85).aspx
+
+    int xScreen = ::GetSystemMetrics(SM_XVIRTUALSCREEN);
+    int yScreen = ::GetSystemMetrics(SM_YVIRTUALSCREEN);
+    int cxScreen = ::GetSystemMetrics(SM_CXVIRTUALSCREEN);
+    int cyScreen = ::GetSystemMetrics(SM_CYVIRTUALSCREEN);
+
+    // make a copy, we need to adjust the comparison
+    //  if it's a maximized window because the OS move the 
+    //  origin of the window by 8 pixes to releive the borders 
+    //  from the monitor for legacy apps
+    int xLeft = left;
+    int xTop = top;
+    int xWidth = width;
+    int xHeight = height;
+
+    if (showCmd == SW_MAXIMIZE) {
+        xLeft += kWindowFrameSize;
+        xTop += kWindowFrameSize;
+        xWidth -= kWindowFrameSize * 2;
+        xHeight -= kWindowFrameSize * 2;
+    }
+
+    // Make sure the window fits inside the virtual screen.
+    // If it doesn't then we let windows decide the window placement
+    if (xLeft < xScreen ||
+        xTop  < yScreen ||
+        xLeft + xWidth > xScreen + cxScreen ||
+        xTop + xHeight > yScreen + cyScreen) {
+
+        // something was off-screen so reposition
+        //  to the default window placement 
+        left   = CW_USEDEFAULT;
+        top    = CW_USEDEFAULT;
+        width  = CW_USEDEFAULT;
```

Comment:
Well, this has to work for both the maximized size / placement and the restored size / placement otherwise you run into that odd case so, IMO, it's better to just let the OS decide initial placement and then let the user move it wherever they want.  

This eliminates incorrectly computing the placement or placing the window on a secondary monitor (in the case they have more than 2 monitors and disconnect one or rearrange monitors.)